File: Amigo.Tenant.UI.Web/src/app/amigotenant/contract/dto/contract-change-term-request.ts
export class ContractChangeTermRequest{
    contractTermType: string | null;
    //Extension
    finalPeriodId: number | null;
    //Modify
    fromPeriodId: number | null;
    //Common
    newTenantId: number | null;
    newDeposit: number;
    newRent: number;
    newHouseId: number | null;
    //Additional Data
    contractId: number | null;
}

File: Amigo.Tenant.Application.DTOs/Requests/Leasing/ContractChangeTermRequest.cs
﻿using Amigo.Tenant.Application.DTOs.Requests.Common;

namespace Amigo.Tenant.Application.DTOs.Requests.Leasing
{
    public class ContractChangeTermRequest : AuditBaseRequest
    {
        public string ContractTermType { get; set; }
        public int? FinalPeriodId { get; set; }
        public int? FromPeriodId { get; set; }

        public int? NewTenantId { get; set; }

        public decimal? NewDeposit { get; set; }
        public decimal? NewRent { get; set; }
        public int? NewHouseId { get; set; }

        public int ContractId { get; set; }
        
    }
}